src/services/user-service.js

```diff
@@ -1,17 +1,17 @@
 import firebase from '../firebase';
 
 const USERS_COLLECTION = 'users';
+const ALL_USERS = firebase.firestore().collection(USERS_COLLECTION);
 
 /**
  * Create/Update a profile
  *
  * @param {string} userId id of the user
- * @param {object} fields an object with fields and their values
+ * @param {object} user an object with fields and their values
  * @return {object} a promise
  */
-const setProfile = (userId, ...fields) => {
-  return firebase.firestore().collection(USERS_COLLECTION).doc(userId).set(
-    ...fields, { merge: true });
+const updateUser = (userId, user) => {
+  return ALL_USERS.doc(userId).set(user);
```

Comment:
`set` operation will overwrite any info that already exists. We need the option `{merge: true}` to prevent overwrites.
Also, we can use the `set` op for both new **inserts** and **updates**